srcalc/src/main/java/gov/va/med/srcalc/service/DefaultCalculationService.java

```diff
@@ -87,7 +94,18 @@ public CalculationResult runCalculation(
     {
         LOGGER.debug("Running calculation with values: {}", variableValues);
         
-        final CalculationResult result = calculation.calculate(variableValues);
+        final boolean firstRun = !calculation.getHistoricalCalculation().isPresent();
+
+        final VistaPerson user = SecurityUtil.getCurrentPrincipal().getVistaPerson();
+        final CalculationResult result = calculation.calculate(variableValues, user);
+        
+        // If it was the first run, save the historical result for metrics.
+        if (firstRun)
+        {
+            // Can't save for now due to double-persist issue when trying to save
+            // SignedResult.
+            LOGGER.debug("Was first run, but saving HistoricalCalc not yet implemented.");
```

Comment:
ASRC-52 doesn't require this functionality, but ASRC-59 does so I will fix it there.